driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java

```diff
@@ -857,5 +870,14 @@ public Select from(TableMetadata table) {
       previousSelection = null;
       return super.from(table);
     }
+
+    @Override
+    public Select from(MaterializedViewMetadata view) {
+      if (previousSelection != null) {
+        addName(previousSelection);
+      }
+      previousSelection = null;
```

Comment:
Why we are setting it back explicitly to `null`? To prevent memory leak?

Reply:
That's a trick to handle aliased selections:
```java
select().column("column1").as("alias1")
```
Internally all selectors are added to the `columnNames` list. But if a column is aliased, it must be wrapped into an `Alias` before (see implementation of `as()`).
`column()` doesn't know yet if its argument will be aliased or not, because that only comes in the `as()` called that is chained after. So instead it stores its argument temporarily in `previousSelection`.
If the next method is `as()`, it will wrap `previousSelection` and add it.
If it's any other method that can be chained after `column`, it adds it directly. `from()` is one of these methods.

`previousSelection` must be reset so that another method doesn't add it again later.